### ISSUE TITLE
Using FFLAGS, etc, in makenek

### ISSUE THREAD

**Issue by RonRahaman** [Nek5000/Nek5000#74]
It would be very useful if makenek honored the following environment variables:
- F77
- FFLAGS
- CC
- CFLAGS
- CPPFLAGS
  makenek.inc could append the user's FFLAGS to whatever other flags it requires.  This is the default behavior for an autotools-derived build system, and I've seen many simple makefiles that do that as well.  

It's a clean way for the users to customize their builds without manually editing makefiles.

**Comment by stgeke** [comment]
This enables us to run `makenek` with customized settings w/o manually editing the file. Note
`makenek` creates a makefile for you. It is not recommended to edit the makefile by hand.

**Comment by RonRahaman** [comment]
Then you should remove this error message from makenek.inc

```
case $F77comp in
...
  *)  echo "ERROR: Unable to detect compiler!"
      echo "        - don't know how to promote datatype REAL to 8 bytes"
      echo "        - don't know how to invoke the C pre-processor (CPP) before compilation"
      echo "       Please edit the makefile and specify the requested compiler flags using the P variable."
      echo ""
      P="<specify your compiler flags here>"
      NOCOMPILE=1
      read;;
esac
```

**Comment by RonRahaman** [comment]
I mean, right now, if the user wants to set their own compiler flags, their only option is, in fact, to manually edit their makefile.  And apparently, that's endorsed by makenek.inc (but I agree, it shouldn't be)

I agree that it would be good to run makenek with user-defined F77, FFLAGS, etc.  However, we shouldn't prohibit users from running make with customized F77, FFLAGS, as well.  It's expected, well-documented behavior in GNU make ([examples](http://stackoverflow.com/questions/2826029/passing-additional-variables-from-command-line-to-make)), and our makefiles should account for that usage.

**Comment by stgeke** [comment]
You can use your own flags. Just add them to CC and F77 in makenek. 
To run make instead of makenek is possible, but requires a major redesign of our build-system.

**Comment by RonRahaman** [comment]
Can we at least separate F77 and FFLAGS in makenek?

**Comment by stgeke** [comment]
You can set generic compiler flags using the G variable in `makenek`

**Comment by adampep** [comment]
When it comes to specifying compilers and flags makenek.inc is not an optimal tool on cray, as all the compilers are wrapped up and named ftn. In this case we have to modify makenek.inc every time we change compiler module. Maybe it would be better to take it out from makenek.inc add single file e.g. system-compiler.h containing all the system specific variables (compilers, flags, commands), that would be used during compilation step. This file would be generated only once for given platform-compiler set and could be easily modified by user. One could make some tool generating this file and set some space for users to exchange their configuration files. Flash code e.g. includes in the distribution the set of configuration files for known systems.

**Comment by stgeke** [comment]
@adampep We fixed the ftn handling a few weeks ago.

**Comment by njansson** [comment]
@stgeke but the current `ftn` fix assumes `PrgEnv-cray`, which depending on the purpose might not be the most optimal compiler on a Cray.

A better way would be to check for `ftn`, if true, then check if either `CRAY_PRGENVCRAY`, `CRAY_PRGENVINTEL`, `CRAY_PRGENVPGI`, `CRAY_PRGENVGNU`... is defined to figure out which backend compiler `ftn` wraps. Also, if the default case assumes Cray, it will not break makenek on older Cray systems (prior to XT).

**Comment by stgeke** [comment]
No, we check what compiler the wrapper (ftn) is using. So it works with Cray, Intel or PGI. See here #17.

**Comment by njansson** [comment]
@stgeke Nice! my bad. In that case I don't see why specifying flags in G wouldn't be sufficient on a Cray? @adampep

**Comment by RonRahaman** [comment]
@stgeke, the G variable in makenek only applies to Fortran compilation, not C compilation.  If the user could set FFLAGS and CFLAGS, we could handle both.

**Comment by stgeke** [comment]
Partly implemented in #223
